File: app/src/app/core/models/grade-item.ts
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

export type GradeItem = {
  id: number;
  cmid: number | null;
  name: string | null;
  type: string;
  module: string | null;
  instance: number;
  number: number | null;
  category: number | null;
  weight: number | null;
  value: number | null;
  minValue: number | null;
  maxValue: number | null;
  isUserLocked: boolean;
  isLocked: boolean;
  isHidden: boolean;
  isOverridden: boolean;
  needsUpdate: boolean;
  isHiddenByDate: boolean;
  submittedTs: number | null;
  gradedTs: number | null;
  status: string | null;
  feedback: string | null;
};

File: api/src/main/kotlin/es/ubu/lsi/ubumonitorweb/domain/GradeItem.kt
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

package es.ubu.lsi.ubumonitorweb.domain

data class GradeItem(
  val id: Int,
  val cmid: Int?,
  val name: String?,
  val type: String,
  val module: String?,
  val instance: Int,
  val number: Int?,
  val category: Int?,
  val weight: Double?,
  val value: Double?,
  val minValue: Double?,
  val maxValue: Double?,
  val isUserLocked: Boolean,
  val isLocked: Boolean,
  val isHidden: Boolean,
  val isOverridden: Boolean,
  val needsUpdate: Boolean,
  val isHiddenByDate: Boolean,
  val submittedTs: Long?,
  val gradedTs: Long?,
  val status: String?,
  val feedback: String?,
)
